src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1125,68 +1126,77 @@ public async Task SetUnusedDatabaseIds()
 
             await ServerStore.EnsureNotPassiveAsync();
 
+            AggregateException ae = null;
+            HashSet<string> unusedIds;
+
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             using (var json = await context.ReadForDiskAsync(RequestBodyStream(), "unused-databases-ids"))
             {
                 var parameters = JsonDeserializationServer.Parameters.UnusedDatabaseParameters(json);
-                if (validate)
-                {
-                    using (var token = CreateHttpRequestBoundTimeLimitedOperationToken(ServerStore.Configuration.Cluster.OperationTimeout.AsTimeSpan))
-                        await ValidateUnusedIdsAsync(parameters.DatabaseIds, database, token.Token);
-                }
+                unusedIds = parameters.DatabaseIds;
+                validate |= parameters.Validate;
+            }
 
-                var command = new UpdateUnusedDatabaseIdsCommand(database, parameters.DatabaseIds, GetRaftRequestIdFromQuery());
-                await ServerStore.SendToLeaderAsync(command);
+            if (validate)
+            {
+                using (var token = CreateHttpRequestBoundTimeLimitedOperationToken(ServerStore.Configuration.Cluster.OperationTimeout.AsTimeSpan))
+                    ae = await ValidateUnusedIdsAsync(unusedIds, database, token.Token);
             }
 
+            var command = new UpdateUnusedDatabaseIdsCommand(database, unusedIds, GetRaftRequestIdFromQuery());
+            await ServerStore.SendToLeaderAsync(command);
+
+            if (ae is not null)
+                throw ae;
```

Comment:
this is wrong... if you have an exception, you still modify the unused ids?

Reply:
Yes,
I add the valid id but throw about the invalid ids, which I don't add

Reply:
Changed